dotCMS/src/enterprise/java/com/dotcms/enterprise/publishing/remote/handler/ContentHandler.java

```diff
@@ -331,14 +333,29 @@ private void handleContents(final Collection<File> contents, final File folderOu
 						// This operation (PUBLISH) publishes/un-publishes a Content that is NOT archived. The un-publishing
 						// is taken care of at the end of this handleContents() method
 						publish(content, folderOut, systemUser, wrapper, isHost,remoteLocalLanguages);
+						Logger.debug(this, "Content saved: " + content.getIdentifier()
+								+ " , inode: " + content.getInode() + " , language: " + content.getLanguageId());
 					} else if (wrapper.getOperation().equals(PushPublisherConfig.Operation.PUBLISH) && isPushedContentArchived) {
 						// This operation (PUBLISH) flags the pushed Content as Archived, WITHOUT deleting it
+						boolean localContentAlreadyExists = APILocator.getContentletAPI().isContentlet(content.getInode());
+						if (!localContentAlreadyExists) {
+							// If the content doesn't exist in the local instance yet, we need to save it first
+							publish(content, folderOut, systemUser, wrapper, isHost, remoteLocalLanguages);
+							Logger.debug(this, "Content saved before archive: " + content.getIdentifier()
+									+ " , inode: " + content.getInode() + " , language: " + content.getLanguageId());
+						}
 						unpublish(content, systemUser, isHost, remoteLocalLanguages, true);
+						Logger.debug(this, "Content archived: " + content.getIdentifier()
```

Comment:
Content deleted? shouldn't be unpublish?

Reply:
I renamed the method to `archiveOrDelete` for clarity.